core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/common/AkkaActorRefMappingService.scala

```diff
@@ -66,17 +68,37 @@ class AkkaActorRefMappingService(actorService: AkkaActorService) extends AmberLo
         }
       }
     }
+    if (toNotifyOnRegistration.contains(id)) {
+      toNotifyOnRegistration(id).foreach { toNotify =>
+        toNotify ! RegisterActorRef(id, ref)
+      }
+      toNotifyOnRegistration.remove(id)
+    }
   }
 
   def retrieveActorRef(id: ActorVirtualIdentity, replyTo: Set[ActorRef]): Unit = {
     if (actorRefMapping.contains(id)) {
       replyTo.foreach { actor =>
         actor ! RegisterActorRef(id, actorRefMapping(id))
       }
-    } else if (actorService.parent != null) {
-      actorService.parent ! GetActorRef(id, replyTo + actorService.self)
+    } else if (actorId != CONTROLLER) {
+      // propagation stops at controller
+      if (!queriedActorVirtualIdentities.contains(id)) {
+        try {
+          actorService.parent ! GetActorRef(id, replyTo + actorService.self)
+          queriedActorVirtualIdentities.add(id)
+        } catch {
+          case e: Throwable =>
+            logger.error(
+              "Failed to fetch actorRef for " + id + " parentRef = " + actorService.parent
+            )
```

Comment:
is this an error or a warning?
if it is an error, should we rethrow it to terminate the execution?

Reply:
this is a warning.